core/src/Data/DataSource/DataSourceMapper.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Harmony\Core\Data\DataSource;
+
+use Harmony\Core\Data\Mapper\Mapper;
+use Harmony\Core\Data\Query\Query;
+
+/**
+ * @template   T
+ * @template   TEntity
+ * @template   TData
+ * @implements GetDataSource<TEntity>
+ * @implements PutDataSource<TEntity>
+ */
+class DataSourceMapper implements
+  GetDataSource,
+  PutDataSource,
+  DeleteDataSource {
+  /**
+   * @param GetDataSource<TData>   $getDataSource
+   * @param PutDataSource<TData>   $putDataSource
+   * @param DeleteDataSource       $deleteDataSource
+   * @param Mapper<TEntity, TData> $entityToDataMapper
+   * @param Mapper<TData, TEntity> $dataToEntityMapper
+   */
+  public function __construct(
+    protected readonly GetDataSource $getDataSource,
+    protected readonly PutDataSource $putDataSource,
+    protected readonly DeleteDataSource $deleteDataSource,
+    protected readonly Mapper $entityToDataMapper,
+    protected readonly Mapper $dataToEntityMapper,
+  ) {
+  }
+
+  /**
+   * @return T
+   */
+  public function get(Query $query): mixed {
+    $toMap = $this->getDataSource->get($query);
+
+    if (!is_array($toMap)) {
```

Comment:
The same I wrote in the repository mapper, you shouldn't need to check if it's an array because T can be `campaign[]` then you have a mapper from CampaignEntity[] to Campaign[] and the mapper will be ArrayMapper, which the one that loops and maps. We need to create this mapper here. As a reference:
https://github.com/mobilejazz/harmony-typescript/blob/master/packages/core/src/repository/mapper/mapper.ts#L149

Reply:
Fixed.